### PR TITLE
Setup github actions for Windows and macOS

### PR DESCRIPTION
**Pull request by hcarty**
Starting without Ubuntu tests because I've been having trouble getting orx's Rebol-based setup process to work on github action's runners.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        gmakeplatform:
+          - 64
+        ocamlversion:
+          - 4.13.x
+
+    steps:
+      - name: Checkout ocaml-orx
+        uses: actions/checkout@v2
+        with:
+          path: ocaml-orx
```
Comment (smorimoto) [inline .github/workflows/main.yml:35]: The problem here is that by changing the configuration of the checkout action, the opam files are extracted outside the default search path.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        working-directory: orx/code/build/windows/gmake
+        run: make config=${{ matrix.gmakeconfig }}${{ matrix.gmakeplatform }}
+
+      - name: Set ORX environment
+        run: echo "ORX=${{ github.workspace }}/orx/code" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
+
+      - name: Setup OCaml
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocamlversion }}
```
Comment (smorimoto) [inline .github/workflows/main.yml:57]: It should work as expected by doing the following:

```yml
opam-local-packages: ocaml-orx/*.opam
```

https://github.com/ocaml/setup-ocaml/blob/master/examples.md#using-glob-patterns-to-filter-local-packages

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+      - name: Install system dependencies
+        working-directory: ocaml-orx
+        run: opam depext conf-pkg-config conf-libffi --with-test
```
Comment (smorimoto) [inline .github/workflows/main.yml]: So now we can remove these lines.

File: .github/workflows/main.yml (hunk 1 above)
Comment (hcarty) [reply .github/workflows/main.yml:35]: Ah of course - that makes perfect sense. Thank you for taking time to look at this!

File: .github/workflows/main.yml (hunk 2 above)
Comment (hcarty) [reply .github/workflows/main.yml:57]: Makes sense - I'll test now!

**Comment by hcarty** [comment]
Thank you again @smorimoto for your help in fixing my issues on Windows! I'll take care of the remaining Windows build issues in a followup PR.

**Comment by smorimoto** [comment]
Anytime!

**Pull request merged**